vulnerability fixes for Sol003

settipalli2F51 opened the pull request:
Fixes #23

settipalli2F51 committed:
Update PackageManagementControllerTest.java

#issue/23

settipalli2F51 committed:
Update pom.xml

#issue/23

swatij-ibm reviewed (changes requested):
@settipalli2F51 

1. Does this even build ? 
<img width="724" alt="Screenshot 2022-01-07 at 13 45 16" src="https://user-images.githubusercontent.com/55978914/148553116-e7a9b4d8-36dc-4775-84d9-fbdc05677735.png">
I don't see a 3.5.0 spring boot available https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-parent

2. I think we shouldn't be commenting out tests, I would prefer the tests to be handled and fixed. @hhellyer  what do you suggest/think on this ?

3. This repo does not have any automated Jenkins etc to check the workability of the new build. Last I recall, when vulnerabilities were fixed, before merging the changes into develop, a test install was run using a locally built image with the required changes. 
Has an installation test with these changes been tested ? 
If not then probably this needs to be included and done before the review and merge to validate the newly built image of sol installs fine.

clloydaccanto reviewed (changes requested):
I agree with the comments left by @swatij-ibm

settipalli2F51 committed:
Update pom.xml

Fixes #23

settipalli2F51 committed:
Update PackageManagementControllerTest.java

changed the advice code to run the Junit test cases .

settipalli2F51 committed:
Update ETSIExceptionHandlingControllerAdvice.java

Changed the advice code to run the Junit test cases .
Spring Boot.2.3.5.RELEASE  version , is not able to parse ProblemDetails object in this particular scenario .So changed the ProblemDetails response as ResponseEntity<String> .

settipalli2F51 committed:
Update ETSIExceptionHandlingControllerAdvice.java

added import statement for Response Entity .

swatij-ibm reviewed (approved):
Thanks for making the changes @settipalli2F51. 
LGTM 👍

clloydaccanto commented on pom.xml:
```diff
@@ -71,6 +71,11 @@
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>30.0-jre</version>
```
can this version number please be moved into the properties section where other version are managed

clloydaccanto commented on src/main/java/com/accantosystems/stratoss/vnfmdriver/web/etsi/ETSIExceptionHandlingControllerAdvice.java:
```diff
@@ -90,8 +90,8 @@ protected ProblemDetails handleNotImplementedException(HttpServletRequest req, N
-    protected ProblemDetails handleResponseTypeNotAcceptableException(HttpServletRequest req, ResponseTypeNotAcceptableException cause) {
-        return defaultHandle("The requested response type was not acceptable", cause, HttpStatus.NOT_ACCEPTABLE);
+    protected ResponseEntity<String> handleResponseTypeNotAcceptableException(HttpServletRequest req, ResponseTypeNotAcceptableException cause) {
+        return new ResponseEntity<>("The requested response type was not acceptable", HttpStatus.NOT_ACCEPTABLE);
```
sorry but I don't think this is the right fix. We can't just change the response type from a ProblemDetails structure (as expected by the ETSI spec) to plain string response, especially as it won't contain the cause as it did previously. Also by not calling defaultHandle we no longer have any of the logging that went on in that method

settipalli2F51 committed:
Reverted the changes in Advice class , changed the springboot and other libraries versions
